syntax = "proto3";

package orderservice;

service OrderService {
  rpc CreateOrder(CreateOrderRequest) returns (CreateOrderResponse);
  rpc OrderStateChanged(OrderStateChangedSubscribeRequest) returns (stream OrderStateChangedUpdate);
}

message CreateOrderRequest {
  int32 item_id = 1;
  bool autostart = 2;
}

message CreateOrderResponse {
  bool success = 1;
  int32 order_id = 2;
}

message OrderStateChangedSubscribeRequest {}

message OrderStateChangedUpdate {
  int32 order_id = 1;
  OrderState state = 2;
}

enum OrderState {
  Created = 0;
  InProduction = 1;
  Completed = 2;
}
